File: usages/python/CarthaPirates.py
import os
import requests

class CarthaPirates(object):

  def __init__(self, idBateau):
    self.monBateau = idBateau
    self.setApiUrl()

  def setApiUrl(self):
    dir = os.path.dirname(os.path.realpath(__file__))
    isDev = os.path.exists("{}/dev".format(dir))
    if isDev:
      self.url = "http://localhost:9001"
    else:
      self.url = "https://carthapirates.fr/api"

  def listerTousLesPorts(self):
    url = self.url + "/ports"
    response = requests.get(url)
    return response.json()

  def deplacerMonBateauVersCoords(self, coords):
    url = self.url + '/bateaux/' + str(self.monBateau)
    params = { "longitude": coords[0], "latitude": coords[1] }
    requests.put(url, params=params)

  def recupererCoordsPort(self, idPort):
    url = self.url + "/ports/" + str(idPort)
    response = requests.get(url)
    return response.json()["geom"]["coordinates"]

  def recupererCoordsMonBateau(self):
    url = self.url + "/bateaux/" +  str(self.monBateau)
    response = requests.get(url)
    return response.json()["geom"]["coordinates"]

  def trouverPortsProchesCoords(self, coords, nombre):
    url = self.url + "/ports/proches/" + str(nombre) + "/" + str(coords[0]) + "/" + str(coords[1])
    response = requests.get(url)
    return response.json()

  def trouverBateauxProchesCoords(self, coords, nombre):
    url = self.url + "/bateaux/proches/" + str(nombre) + "/" + str(coords[0]) + "/" + str(coords[1])
    response = requests.get(url)
    return response.json()

  def supprimerLesTrajetsDeMonBateau(self):
    url = self.url + "/trajets/" + str(self.monBateau)
    requests.delete(url)

  def rentrerMonBateau(self):
    url = self.url + '/bateaux/' + str(self.monBateau) + '/rentrer'
    requests.put(url)

File: usages/javascript/CarthaPirates.js
class CarthaPirates {
    
    constructor(idBateau) {
        this.monBateau = idBateau
        this.init = false
        this.setApiUrl()
    }

    setApiUrl() {
        const isDev = true // FIX ME
        if (isDev) {
            this.url = 'http://localhost:9001'
        } else {
            this.url = 'https://carthapirates.fr/api'
        }
    }

    async listerTousLesPorts() {
        const url = this.url + '/ports'
        const response = await fetch(url)
        return await response.json()
    }

    async deplacerMonBateauVersCoords(coords) {
        const params = new URLSearchParams({ longitude: coords[0], latitude: coords[1] })
        const url = this.url + '/bateaux/' + this.monBateau + '?' + params
        return fetch(url, { method: 'PUT'})
    }

    async recupererCoordsPort(idPort) {
        const url = this.url + '/ports/' + idPort
        const response = await fetch(url)
        return await response.json()
    }

    async recupererCoordsMonBateau() {
        const url = this.url + '/bateaux/' + this.monBateau
        const response = await fetch(url)
        return await response.json()
    }

    async trouverPortsProchesCoords(coords, nombre) {
        const params = new URLSearchParams({ longitude: coords[0], latitude: coords[1] })
        const url = this.url + '/ports/' + nombre + '/proches?' + params
        const response = await fetch(url)
        return await response.json()
    }

    async trouverBateauxProchesCoords(coords, nombre) {
        const params = new URLSearchParams({ longitude: coords[0], latitude: coords[1] })
        const url = this.url + '/bateaux/' + nombre + '/proches?' + params
        const response = await fetch(url)
        return await response.json()
    }

    async rentrerMonBateau() {
        const url = this.url + '/bateaux/' + this.monBateau + '/rentrer'
        return fetch(url, { method: 'PUT'})
    }
}